webservices/react-testing.md

```diff
@@ -0,0 +1,431 @@
+        className={stylesImage}
+      />
+    </div>
+    <div className={styles.rightSection}>
+      <div className={styles.rightSectionTopHalf}>
+        <div className={styles.timeSection}>
+          <span className={styles.timeText}>{timeText}</span>
+          <span className={styles.dateTimeText}>{dateTimeText}</span>
+        </div>
+        {testMeal || (
+          <div className={styles.portionSection}>
+            <Portion portion={portion} />
+          </div>
+        )}
+      </div>
+      <div className={styles.noteSection}>
+        <span className={stylesNoteText}>{noteText}</span>
+      </div>
+    </div>
+  </div>
+);
+
+MealEvent.propTypes = {
+  dateTimeText: PropTypes.string.isRequired,
+  flagged: PropTypes.bool.isRequired,
+  mealEventImgOnError: PropTypes.func.isRequired,
+  noteText: PropTypes.string.isRequired,
+  portion: PropTypes.string,
+  srcMealEvent: PropTypes.string.isRequired,
+  stylesImage: PropTypes.string.isRequired,
+  stylesNoteText: PropTypes.string.isRequired,
+  testMeal: PropTypes.bool.isRequired,
+  timeText: PropTypes.string.isRequired
+};
+
+export default enhance(MealEvent);
+```
+
+You want to write tests that assert that the following are rendered if applicable:
+
+* Text
+* Event Handlers
+* Styles
+* Image Tags
+* Components
+
+These may be present, absent, or change dynamically based on the given props. Here is an example test file:
+
+*__tests__/index.js*
+
+```js
+import React from "react";
+import { shallow } from "enzyme";
+import { MealEvent, text } from "../index";
+import styles from "../styles.module.scss";
+import IconFlag from "components/UI/IconFlag";
+import Portion from "components/UI/Portion";
+
+describe("MealEvent", () => {
+  let props, wrapper;
+
+  beforeEach(() => {
+    props = {
+      dateTimeText: "9:37 AM",
+      flagged: true,
+      mealEventImgOnError: jest.fn(),
+      noteText: "I am a note",
+      onClick: jest.fn(),
+      portion: "Regular",
+      srcMealEvent: "http://via.placeholder.com/350x150",
+      stylesImage: "imageMealEvent",
+      stylesNoteText: "noteTextAbsent",
+      testMeal: false,
+      timeText: "Log Time: 10:39 AM on Jun 1, 2017"
+    };
+    wrapper = shallow(<MealEvent {...props} />);
+  });
+
+  it("renders dateTimeText", () => {
+    expect(wrapper).toIncludeText(props.dateTimeText);
+  });
+
+  it("renders noteText", () => {
+    expect(wrapper).toIncludeText(props.noteText);
+  });
+
+  it("renders timeText", () => {
+    expect(wrapper).toIncludeText(props.timeText);
+  });
+
+  it("handles click events", () => {
+    wrapper.simulate("click");
+
+    expect(props.onClick).toHaveBeenCalled();
+  });
+
+  it("renders a span with className of stylesNoteText", () => {
+    const element = wrapper
+      .find("span")
+      .filterWhere(n => n.props().className === props.stylesNoteText);
+
+    expect(element).toBePresent();
+  });
+
+  it("renders img whose src is srcMealEvent", () => {
```

Comment:
Would just add a note about checking the `mealEventImgOnError` function to the description since it seems out of place in the test, i.e. "renders img whose src is srcMealEvent and calls the correct error function"